src/components/app-configure/access/CreateDelegate.vue

```diff
@@ -0,0 +1,170 @@
+<script lang="ts" setup>
+import { ref } from 'vue'
+
+import VButton from '@/components/lib/VButton/VButton.vue'
+import VCard from '@/components/lib/VCard/VCard.vue'
+import VDropdown from '@/components/lib/VDropdown/VDropdown.vue'
+import VOverlay from '@/components/lib/VOverlay/VOverlay.vue'
+import VSeperator from '@/components/lib/VSeperator/VSeperator.vue'
+import VStack from '@/components/lib/VStack/VStack.vue'
+import VTextField from '@/components/lib/VTextField/VTextField.vue'
+import { useToast } from '@/components/lib/VToast'
+import { createDelegate } from '@/services/gateway.service'
+import { useAppId } from '@/use/getAppId'
+
+const emit = defineEmits(['close', 'generateKey', 'created'])
+
+const props = defineProps({
+  delegateKeys: {
+    type: Array,
+    required: true,
+  },
+})
+
+const loader = ref({
+  show: false,
+  message: '',
+})
+
+const toast = useToast()
+const appId = useAppId()
+const delegateName = ref('')
+const selectedDelegateAddress = ref('')
+
+function showLoader(message: string) {
+  loader.value.show = true
+  loader.value.message = message
+}
+
+function hideLoader() {
+  loader.value.show = false
+  loader.value.message = ''
+}
+
+async function onSubmit() {
+  if (!delegateName.value.length || !selectedDelegateAddress.value.length) {
+    toast.error('Please fill all required values')
+  }
+  showLoader('Created Delegate...')
+  try {
+    const payload = {
+      name: delegateName.value,
+      address: selectedDelegateAddress.value,
+      permissions: ['Dowload', 'Reshare'],
+    }
+    const { data } = await createDelegate(appId, payload)
```

Comment:
Need to perform smartcontract operation if this is successful

Need to call `grantDelegate` function from `smart-contract.service.ts`

Reply:
ok